src/Component/DoxenControl.php

```diff
@@ -48,24 +48,22 @@ class DoxenControl extends Control
 
 	public function __construct(DocTree $tree, ?Config $config = null)
 	{
-		parent::__construct();
 		$this->tree = $tree;
 		$this->config = $config ?: new Config();
+		$this->setMonitor();
 	}
 
+
 	/**
-	 * @param IPresenter $presenter
-	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
+	 * Set monitor in component
 	 */
-	protected function attached($presenter): void
+	private function setMonitor()
 	{
-		parent::attached($presenter);
-
-		if ($presenter instanceof Presenter) {
+		$this->monitor(Control::class, function (){
```

Comment:
It should be Presenter, not Control. Some of the parts connected by events depend on it's availability